src/main/java/io/toadlabs/numeralping/mixin/ServerEntryMixin.java

```diff
@@ -1,46 +1,46 @@
 import net.minecraft.client.MinecraftClient;
-import net.minecraft.client.font.TextRenderer;
 import net.minecraft.client.gui.DrawContext;
-import net.minecraft.client.gui.screen.multiplayer.*;
+import net.minecraft.client.gui.screen.multiplayer.MultiplayerServerListWidget;
 import net.minecraft.client.network.ServerInfo;
-import net.minecraft.text.*;
+import net.minecraft.text.Text;
+import net.minecraft.text.TranslatableTextContent;
 import net.minecraft.util.Identifier;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.ModifyArgs;
+import org.spongepowered.asm.mixin.injection.Redirect;
+import org.spongepowered.asm.mixin.injection.invoke.arg.Args;
+
+import java.util.List;
 
 // a priority of 2000 means it will apply later
 // this is combined with `require = 0` to allow other mods to apply more integral functionality first without the game crashing
 @Mixin(value = MultiplayerServerListWidget.ServerEntry.class, priority = 0)
 public class ServerEntryMixin {
 
-	@Redirect(method = "render", at = @At(value = "INVOKE", target = "Lnet/minecraft/client/gui/DrawContext;drawText(Lnet/minecraft/client/font/TextRenderer;Lnet/minecraft/text/Text;IIIZ)I", ordinal = 0))
-	public int shiftText(DrawContext instance, TextRenderer renderer, Text text, int x, int y, int color, boolean shadow) {
+	@ModifyArgs(method = "render", at = @At(value = "INVOKE", target = "Lnet/minecraft/client/gui/DrawContext;drawText(Lnet/minecraft/client/font/TextRenderer;Lnet/minecraft/text/Text;IIIZ)I", ordinal = 0))
+	public void shiftText(Args args) {
 		NumeralConfig config = NumeralConfig.instance();
 
 		if (config.serverList) {
-			x += 10 - renderer.getWidth(getPingText(config, server.ping));
+			args.set(2, ((int) args.get(2)) + 10 - client.textRenderer.getWidth(getPingText(config, server.ping)));
 		}
-
-		return instance.drawText(renderer, text, x, y, color, false);
 	}
 
 	// hide the tooltip if it's redundant
-	@Redirect(method = "render", at = @At(value = "INVOKE", target = "Lnet/minecraft/client/gui/screen/multiplayer/MultiplayerScreen;setMultiplayerScreenTooltip(Ljava/util/List;)V"), require = 0)
-	public void setTooltip(MultiplayerScreen screen, List<Text> tooltip) {
-		if (tooltip != null && tooltip.size() == 1 && NumeralConfig.instance().serverList
-				&& tooltip.get(0).getContent() instanceof TranslatableTextContent translatable
-				&& translatable.getKey().equals("multiplayer.status.ping")) {
-			tooltip = null;
+	@ModifyArgs(method = "render", at = @At(value = "INVOKE", target = "Lnet/minecraft/client/gui/screen/multiplayer/MultiplayerScreen;setMultiplayerScreenTooltip(Ljava/util/List;)V"), require = 0)
+	public void setTooltip(Args args) {
+		@SuppressWarnings("unchecked") final var tooltip = (List<Text>) args.get(0);
```

Comment:
```suggestion
		var tooltip = (List<Text>) args.get(0);
```

Reply:
Done

Reply:
I don't really want a lot of annotations :P
It's just a harmless warning